File: Problems/problem1.js
const array1 = [1, 2, 3, 4];
const array2 = [5, 6, 7, 8, 9];
const array3 = [1, 9, 10];

function compArrays(arrayA, arrayB) {
  for (let i = 0; i < arrayA.length; i++) {
    for (let j = 0; j < arrayB.length; j++) {
      if (arrayA[i] != arrayB[j]) {
        break;
      } else {
        return true;
      }
    }
  }
  return false;
}

function compArrays2(arrayA, arrayB) {
  let map = {};
  for (let i = 0; i < arrayA.length; i++) {
    const iterable = arrayA[i];
    map[iterable] = true;
  }
  for (let i = 0; i < arrayB.length; i++) {
    if (map[arrayB[i]]) {
      return true;
    }
  }
  return false;
  console.log(map)
}

let result = compArrays2(array2, array3)
console.log(result) // bigO is O(n^2)

File: Problems/problem1.py
array1 = [1, 2, 3, 4]
array2 = [5, 6, 7, 8, 9]
array3 = [1, 9, 10]

def compArrays(arrayA, arrayB):
  for num1 in arrayA:
      for num2 in arrayB:
          if num1 != num2:
            break
          else: 
            return True
  return False 

result = compArrays(array1, array3)
print(result)

def compArrays2(arrayA, arrayB):
  map = {}
  for num in arrayA:
    map[num] = True
  for num in arrayB:
    try:
       map[num]
    except:
      pass
    else:
       return True
  return False


result2 = compArrays2(array1, array3)
print(result2)